test/e2e/fixtures/DataGrid/SelectEditCell.tsx

```diff
@@ -0,0 +1,60 @@
+import * as React from 'react';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
+import { DataGrid, GridCellParams } from '@material-ui/data-grid';
+
+const SelectEditInputCell = (props: GridCellParams) => {
+  const { id, value, api, field } = props;
+
+  const handleChange = (event) => {
+    const editProps = { value: event.target.value };
+    api.commitCellChange({ id, field, props: editProps });
+    api.setCellMode(id, field, 'view');
+  };
+
+  return (
+    <Select value={value} onChange={handleChange}>
```

Comment:
How about we open the select as soon as it enters in edit mode? You can find this UX on AG Grid or Airtable. It feels like it currently requires two extra clicks that could be save in the majority of the cases.

I first saw it in test/e2e/index.test.ts.

Reply:
Updated the e2e test and also the other PR that uses a Select.